### PR TITLE
BUG: Fix lxml import in show_versions

### LINKED ISSUE #23934: show_versions fails on master with latest lxml
#### Code Sample, a copy-pastable example if possible

```python
[ins] In [1]: import pandas as pd
p
[ins] In [2]: pd.show_versions()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-3d232a07e144> in <module>
----> 1 pd.show_versions()

~/workspace/pandas/pandas/util/_print_versions.py in show_versions(as_json)
    107             else:
    108                 mod = importlib.import_module(modname)
--> 109             ver = ver_f(mod)
    110             deps_blob.append((modname, ver))
    111         except ImportError:

~/workspace/pandas/pandas/util/_print_versions.py in <lambda>(mod)
     86         ("xlwt", lambda mod: mod.__VERSION__),
     87         ("xlsxwriter", lambda mod: mod.__version__),
---> 88         ("lxml", lambda mod: mod.etree.__version__),
     89         ("bs4", lambda mod: mod.__version__),
     90         ("html5lib", lambda mod: mod.__version__),

AttributeError: module 'lxml' has no attribute 'etree'
```

#### Expected Output

This seems to work OK on 0.23.4 (output below).

Let me know if there are other potential factors you'd like me to check

#### Output of ``pd.show_versions()``

<details>

[ins] In [5]: pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.1.final.0
python-bits: 64
OS: Darwin
OS-release: 18.0.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.4
pytest: 3.9.2
pip: 18.1
setuptools: 40.6.2
Cython: 0.28.5
numpy: 1.15.2
scipy: None
pyarrow: None
xarray: 0.10.9
IPython: 7.1.1
sphinx: None
patsy: 0.5.0
dateutil: 2.7.5
pytz: 2018.5
blosc: None
bottleneck: 1.2.1
tables: None
numexpr: 2.6.8
feather: None
matplotlib: 3.0.0
openpyxl: None
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: None
lxml: 4.2.5
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.12
pymysql: None
psycopg2: None
jinja2: 2.8.1
s3fs: None
fastparquet: None
pandas_gbq: 0.6.1+2.gd98c621
pandas_datareader: 0.7.0
</details>

### PR DESCRIPTION
**Pull request by nixphix**
Fixed lxml import issue

- [x] closes #23934
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by TomAugspurger — APPROVED** [review]

**Comment by jreback** [comment]
can you add a test for this? 

somethign like

``subprocess.check_call([sys.executable, "-c", "import pandas; pandas.show_versions()"])``

would be enough

**Comment by nixphix** [comment]
@jreback Added test case

File: pandas/tests/util/test_util.py [hunk 1]
```diff
@@ -530,3 +531,10 @@ def test_safe_import(monkeypatch):
     monkeypatch.setitem(sys.modules, mod_name, mod)
     assert not td.safe_import(mod_name, min_version="2.0")
     assert td.safe_import(mod_name, min_version="1.0")
+
+
+def test_lxml_import():
```
Comment (jreback) [inline pandas/tests/util/test_util.py]: can you rename to ``test_show_versions()`` and the issue number is not needed here. maybe just a comment that this test validations that show_versions() is accessing version info correctly.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/util/test_util.py (hunk 1 above)
Comment (TomAugspurger) [reply pandas/tests/util/test_util.py]: FWIW, I don't think this test is necessary. It wouldn't have caught the original bug, unless the environment had lxml but not openpyxl.

We also should be running `show_versions` as part of every build already right? Just for the output in the CI logs?

**Comment by jreback** [comment]
hmm then why didn’t our build fail?

**Comment by TomAugspurger** [comment]
Possibly we didn't have an environment with lxml but not openpyxl.

File: pandas/tests/util/test_util.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/util/test_util.py]: ok, @TomAugspurger is right here, let's just remove this test entirely.

Comment (nixphix) [reply pandas/tests/util/test_util.py]: @jreback removed the test case

**Pull request merged**